### PR TITLE
Feat/supabase

### PR DESCRIPTION
**Pull request by saltcod**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,8 +22,6 @@
     "@radix-ui/react-switch": "^1.1.0",
     "@radix-ui/react-tooltip": "^1.1.2",
     "@supabase/ssr": "^0.5.1",
-    "@vercel/analytics": "^1.3.1",
```
Comment (saltcod) [inline package.json:25]: not in use right now, but also not sure if we want to do something with it

File: lib/chat/actions.tsx [hunk 2]
```diff
@@ -127,22 +127,22 @@ async function submitUserMessage(content: string) {
   let textNode: undefined | React.ReactNode
 
   const result = await streamUI({
-    model: openai('gpt-3.5-turbo'),
+    model: openai('gpt-4o-mini'),
```
Comment (jgoux) [inline lib/chat/actions.tsx:130]: Like @MildTomato said with the GitHub auth, I don't know if we should touch features not related to Supabase here.

**Review by jgoux — APPROVED** [review]

File: lib/chat/actions.tsx (hunk 2 above)
Comment (MildTomato) [reply lib/chat/actions.tsx:130]: Agreed. Don't change.

File: README.md [hunk 3]
```diff
@@ -31,30 +32,40 @@
...
+> Note: You should not commit your `.env.local` file or it will expose secrets that will allow others to control access to your various OpenAI and authentication provider accounts.
 
 1. Install Vercel CLI: `npm i -g vercel`
 2. Link local instance with Vercel and GitHub accounts (creates `.vercel` directory): `vercel link`
 3. Download your environment variables: `vercel env pull`
 
+Next, setup your Supabase project to store your Chat History.
+
+1. In `/supabase/migrations/20240916140519_init.sql`, you'll find a SQL script to setup your database tables in Supabase
+2. Copy this script, and visit the [SQL Editor](https://supabase.com/dashboard/project/zlarqpgkbjqgrvuquugu/sql/new) of the project you just created.
```
Comment (MildTomato) [inline README.md:64]: Needs to be ..

Go to Vercel Storage, click on "open in supabase", go to SQL editor ...

File: README.md [hunk 4]
```diff
@@ -31,30 +32,40 @@
...
 
-## Creating a KV Database Instance
+## Store your Chat History in Supabase
 
-Follow the steps outlined in the [quick start guide](https://vercel.com/docs/storage/vercel-kv/quickstart#create-a-kv-database) provided by Vercel. This guide will assist you in creating and configuring your KV database instance on Vercel, enabling your application to interact with it.
+Go to [supabase.com](https://supabase.com/) and click "Start your project" in the top right to create a new project. When your project is created, grab your api keys. Click the Connect button in the top right of the project homepage -> App Frameworks -> Next.js. Copy the contents of the `.env.local` file with the api keys and add this to your local `.env.local` file.
```
Comment (MildTomato) [inline README.md:45]: This seems wrong. We should be assuming they are using Vercel Storage

**Pull request merged**